server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -269,6 +269,17 @@ public static IndexMergePolicy fromString(String text) {
         Property.IndexScope
     );
 
+    /**
+     * Index setting describing the maximum number of nested scopes in queries and documents.
```

Comment:
This limit should not apply to documents, right? We have limits for nested mappings separateretly

Reply:
Right.  Looks like `index.mapping.depth.limit` is applied for docs index time. Should we sync/merge these two settings? So far I remove `and documents` from javadoc.

Reply:
> Should we sync/merge these two settings?

This is a good question, do you think the query and document limits should be the same? I am not sure since queries could include the conditional logic which increases nesting factor